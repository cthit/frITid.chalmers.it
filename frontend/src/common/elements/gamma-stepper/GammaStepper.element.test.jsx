import React from "react";
import { shallow } from "enzyme";
import GammaStepper from "./";

describe("<GammaStepper/>", () => {
  test("Shallow render of <GammaStepper/>", () => {
    const wrapper = shallow(
      <GammaStepper
        activeStep={3}
        steps={[
          {
            text: "1",
            element: <h2>asdf</h2>
          },
          {
            text: "Det andra steget",
            element: <h2>second</h2>
          },
          {
            text: "Step 3",
            element: <h3>asdffdas</h3>
          }
        ]}
        finishedElement={<h1>Done</h1>}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
