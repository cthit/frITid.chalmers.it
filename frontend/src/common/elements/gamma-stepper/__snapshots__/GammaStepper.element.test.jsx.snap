// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GammaStepper/> Shallow render of <GammaStepper/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaStepper
    activeStep={3}
    finishedElement={
        <h1>
            Done
        </h1>
    }
    steps={
        Array [
            Object {
              "element": <h2>
                asdf
        </h2>,
              "text": "1",
            },
            Object {
              "element": <h2>
                second
        </h2>,
              "text": "Det andra steget",
            },
            Object {
              "element": <h3>
                asdffdas
        </h3>,
              "text": "Step 3",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Stepper)
          activeStep={3}
          alternativeLabel={true}
>
          <WithStyles(Step)>
                    <WithStyles(StepLabel)>
                              1
                    </WithStyles(StepLabel)>
          </WithStyles(Step)>
          <WithStyles(Step)>
                    <WithStyles(StepLabel)>
                              Det andra steget
                    </WithStyles(StepLabel)>
          </WithStyles(Step)>
          <WithStyles(Step)>
                    <WithStyles(StepLabel)>
                              Step 3
                    </WithStyles(StepLabel)>
          </WithStyles(Step)>
</WithStyles(Stepper)>,
        <div>
          <h1>
                    Done
          </h1>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeStep": 3,
          "alternativeLabel": true,
          "children": Array [
            <WithStyles(Step)>
              <WithStyles(StepLabel)>
                            1
              </WithStyles(StepLabel)>
</WithStyles(Step)>,
            <WithStyles(Step)>
              <WithStyles(StepLabel)>
                            Det andra steget
              </WithStyles(StepLabel)>
</WithStyles(Step)>,
            <WithStyles(Step)>
              <WithStyles(StepLabel)>
                            Step 3
              </WithStyles(StepLabel)>
</WithStyles(Step)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(StepLabel)>
                1
</WithStyles(StepLabel)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Det andra steget",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(StepLabel)>
                Det andra steget
</WithStyles(StepLabel)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Det andra steget",
              },
              "ref": null,
              "rendered": "Det andra steget",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Step 3",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(StepLabel)>
                Step 3
</WithStyles(StepLabel)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Step 3",
              },
              "ref": null,
              "rendered": "Step 3",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Done
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Done",
          },
          "ref": null,
          "rendered": "Done",
          "type": "h1",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Stepper)
            activeStep={3}
            alternativeLabel={true}
>
            <WithStyles(Step)>
                        <WithStyles(StepLabel)>
                                    1
                        </WithStyles(StepLabel)>
            </WithStyles(Step)>
            <WithStyles(Step)>
                        <WithStyles(StepLabel)>
                                    Det andra steget
                        </WithStyles(StepLabel)>
            </WithStyles(Step)>
            <WithStyles(Step)>
                        <WithStyles(StepLabel)>
                                    Step 3
                        </WithStyles(StepLabel)>
            </WithStyles(Step)>
</WithStyles(Stepper)>,
          <div>
            <h1>
                        Done
            </h1>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeStep": 3,
            "alternativeLabel": true,
            "children": Array [
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                                1
                </WithStyles(StepLabel)>
</WithStyles(Step)>,
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                                Det andra steget
                </WithStyles(StepLabel)>
</WithStyles(Step)>,
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                                Step 3
                </WithStyles(StepLabel)>
</WithStyles(Step)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(StepLabel)>
                  1
</WithStyles(StepLabel)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "1",
                },
                "ref": null,
                "rendered": "1",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Det andra steget",
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(StepLabel)>
                  Det andra steget
</WithStyles(StepLabel)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Det andra steget",
                },
                "ref": null,
                "rendered": "Det andra steget",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Step 3",
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(StepLabel)>
                  Step 3
</WithStyles(StepLabel)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Step 3",
                },
                "ref": null,
                "rendered": "Step 3",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Done
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Done",
            },
            "ref": null,
            "rendered": "Done",
            "type": "h1",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
